instrumentation/log4j/log4j-appender-2.17/library/src/test/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/AbstractOpenTelemetryAppenderTest.java

```diff
@@ -255,12 +262,35 @@ void logStructuredDataMessage() {
                     .hasInstrumentationScope(instrumentationScopeInfo)
                     .hasBody("a message")
                     .hasAttributesSatisfyingExactly(
-                        equalTo(
-                            ThreadIncubatingAttributes.THREAD_NAME,
-                            Thread.currentThread().getName()),
-                        equalTo(
-                            ThreadIncubatingAttributes.THREAD_ID, Thread.currentThread().getId()),
-                        equalTo(stringKey("log4j.map_message.key1"), "val1"),
-                        equalTo(stringKey("log4j.map_message.key2"), "val2")));
+                        addLocationAttributes(
+                            "logStructuredDataMessage",
+                            equalTo(THREAD_NAME, Thread.currentThread().getName()),
+                            equalTo(THREAD_ID, Thread.currentThread().getId()),
+                            equalTo(stringKey("log4j.map_message.key1"), "val1"),
+                            equalTo(stringKey("log4j.map_message.key2"), "val2"))));
+  }
+
+  private static List<AttributeAssertion> addLocationAttributes(
+      String methodName, AttributeAssertion... assertions) {
+    return addLocationAttributes(AbstractOpenTelemetryAppenderTest.class, methodName, assertions);
+  }
+
+  protected static List<AttributeAssertion> addLocationAttributes(
+      Class<?> testClass, String methodName, AttributeAssertion... assertions) {
+    String selector = System.getProperty("Log4j2.contextSelector");
+    boolean async = selector != null && selector.endsWith("AsyncLoggerContextSelector");
+    if (async) {
```

Comment:
I have not found the test using an asynch appender. Could you please give me a pointer?

The Readme should perhaps explain something about the asynch appenders?

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/29da88e7cf6b8898805c0c8ddc8095477d475ce6/instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts#L48-L54

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/99f1f827a09aa5e15407746df09bbf247df605bf/instrumentation/log4j/log4j-appender-2.17/library/src/test/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/AbstractOpenTelemetryAppenderTest.java#L283 replies to my second point.